lib/phoenix/socket/serializers/v2_json_serializer.ex

```diff
@@ -2,15 +2,58 @@ defmodule Phoenix.Socket.V2.JSONSerializer do
   @moduledoc false
   @behaviour Phoenix.Socket.Serializer
 
+  @push 0
+  @reply 1
+  @broadcast 2
+
   alias Phoenix.Socket.{Broadcast, Message, Reply}
 
   @impl true
+  def fastlane!(%Broadcast{payload: {:binary, data}} = msg) do
+    topic_size = byte_size(msg.topic)
+    event_size = byte_size(msg.event)
+
+    bin = <<
+      @broadcast::size(8),
+      topic_size::size(8),
+      event_size::size(8),
+      msg.topic::binary-size(topic_size),
+      msg.event::binary-size(event_size),
+      data::binary
+    >>
+
+    {:socket_push, :binary, bin}
+  end
+
   def fastlane!(%Broadcast{} = msg) do
```

Comment:
we should probably validate that the payload is a `map` and print a nice error message if it isn't (now that we removed the guard from the channel funs).